### PR TITLE
add %o: obfuscated remote IP address

### PR DESCRIPTION
**Pull request by jjYBdx4IL**
add %o: obfuscated remote IP address, removing the
last IPv4 byte or anything after the second colon (IPv6),
ie. "1.2.3." or "fe08:2:"

Also added the log patterns commonobf and combinedobf where instead of the remote host the obfuscated remote IP gets logged.

Purpose: better support for data privacy laws in the EU.

Tested against WildFly 15.0.1.

### CODE REVIEW

**Comment by undertow-pull-request** [comment]
Can one of the admins verify this patch?

**Comment by stuartwdouglas** [comment]
This is ok to test

**Comment by undertow-pull-request** [comment]
**Windows** [Build 2659](https://ci.wildfly.org/viewLog.html?buildId=138866&buildTypeId=UNDERTOW_MasterWindows) outcome was **FAILURE** using a merge of d048eafe2f497c1ed6885dcf94958ae79c1678a8
Summary: Tests failed: 42 (38 new), passed: 3319, ignored: 709 Build time: 00:43:14
### Failed tests
```
io.undertow.server.InvalidHtpRequestTestCase.testMultipleTransferEncoding: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.InvalidHtpRequestTestCase.testMultipleTransferEncoding(InvalidHtpRequestTestCase.java:128)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.handlers.ChunkedRequestTrailersTestCase.testChunkedRequestsWithTrailers: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at java.net.Socket.connect(Socket.java:538)
	at java.net.Socket.<init>(Socket.java:434)
	at java.net.Socket.<init>(Socket.java:244)
	at io.undertow.server.handlers.ChunkedRequestTrailersTestCase.testChunkedRequestsWithTrailers(ChunkedRequestTrailersTestCase.java:115)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.handlers.FixedLengthRequestTestCase.testFixedLengthRequest: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.handlers.FixedLengthRequestTestCase.testFixedLengthRequest(FixedLengthRequestTestCase.java:97)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.MaxRequestSizeTestCase.testMaxRequestEntitySize: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.MaxRequestSizeTestCase.testMaxRequestEntitySize(MaxRequestSizeTestCase.java:115)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.handlers.DateHandlerTestCase.testDateHandler: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.handlers.DateHandlerTestCase.testDateHandler(DateHandlerTestCase.java:52)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.handlers.ExceptionHandlerTestCase.testAttachException: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.handlers.ExceptionHandlerTestCase.testAttachException(ExceptionHandlerTestCase.java:166)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.HttpServerExchangeTestCase.testHttpServerExchange: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.HttpServerExchangeTestCase.testHttpServerExchange(HttpServerExchangeTestCase.java:66)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.handlers.JDBCLogDatabaseTestCase.testLogLotsOfThreadsToDatabase: java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.net.SocketTimeoutException: Read timed out
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at io.undertow.server.handlers.JDBCLogDatabaseTestCase.testLogLotsOfThreadsToDatabase(JDBCLogDatabaseTestCase.java:202)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)
Caused by: java.lang.RuntimeException: java.net.SocketTimeoutException: Read timed out
	at io.undertow.server.handlers.JDBCLogDatabaseTestCase$2.run(JDBCLogDatabaseTestCase.java:194)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at org.apache.http.impl.io.AbstractSessionInputBuffer.fillBuffer(AbstractSessionInputBuffer.java:161)
	at org.apache.http.impl.io.SocketInputBuffer.fillBuffer(SocketInputBuffer.java:82)
	at org.apache.http.impl.io.AbstractSessionInputBuffer.readLine(AbstractSessionInputBuffer.java:278)
	at org.apache.http.impl.conn.LoggingSessionInputBuffer.readLine(LoggingSessionInputBuffer.java:123)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
	at org.apache.http.impl.AbstractHttpClientConnection.receiveResponseHeader(AbstractHttpClientConnection.java:286)
	at org.apache.http.impl.conn.DefaultClientConnection.receiveResponseHeader(DefaultClientConnection.java:257)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.receiveResponseHeader(ManagedClientConnectionImpl.java:207)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
	at org.apache.http.impl.client.DefaultRequestDirector.tryExecute(DefaultRequestDirector.java:684)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:486)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.handlers.JDBCLogDatabaseTestCase$2.run(JDBCLogDatabaseTestCase.java:188)
	... 5 more


io.undertow.server.InvalidHtpRequestTestCase.testContentLengthAndTransferEncoding: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.InvalidHtpRequestTestCase.testContentLengthAndTransferEncoding(InvalidHtpRequestTestCase.java:114)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.handlers.FixedLengthResponseTestCase.sendHttpRequest: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.handlers.FixedLengthResponseTestCase.sendHttpRequest(FixedLengthResponseTestCase.java:85)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)


io.undertow.server.InvalidHtpRequestTestCase.testInvalidCharacterInMethod: java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:121)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.undertow.server.InvalidHtpRequestTestCase.testInvalidCharacterInMethod(InvalidHtpRequestTestCase.java:73)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)



##### there are 32 more failed tests, see build details
```

**Comment by jstourac** [comment]
@jjYBdx4IL hi, could you please create a tracking jira in Undertow project [here](https://issues.jboss.org/projects/UNDERTOW/issues/UNDERTOW-1492?filter=allopenissues) and ammend description with the jira identifier as is done in other commits: `[UNDERTOW-XXXX] commit title`? Let me know if for any reason you cannot create such jira, I can create one for you.

**Comment by jjYBdx4IL** [comment]
Done.

https://issues.jboss.org/browse/UNDERTOW-1495

**Comment by jjYBdx4IL** [comment]
fyi: the build on Windows seems to have been broken before my changes, already.

btw, since there are quite a few forks on github: I put up a travis config that builds on travis CI public linux and windows servers: https://javagc.leponceau.org/2019/02/running-javamavenkotlingroovysbtscalagr.html

tell me if you want me to add the public travis config to the repo.

File: core/src/main/java/io/undertow/attribute/RemoteObfuscatedIPAttribute.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
```
Comment (fl4via) [inline core/src/main/java/io/undertow/attribute/RemoteObfuscatedIPAttribute.java]: Please chnage to 2019

File: core/src/test/java/io/undertow/util/NetworkUtilsAddressObfuscationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package io.undertow.util;
```
Comment (fl4via) [inline core/src/test/java/io/undertow/util/NetworkUtilsAddressObfuscationTestCase.java]: Please add header

**Comment by fl4via** [comment]
Retest this please

**Comment by jjYBdx4IL** [comment]
license headers fixed.

**Comment by undertow-pull-request** [comment]
**Windows** [Build 2668](https://ci.wildfly.org/viewLog.html?buildId=140321&buildTypeId=UNDERTOW_MasterWindows) outcome was **FAILURE** using a merge of f6c6966bd194489f9adf1fbb2385e08ad607e7fc
Summary: Tests failed: 2 (2 new), passed: 3373, ignored: 709 Build time: 00:57:01
### Failed tests
```
io.undertow.servlet.test.session.CrossContextServletSessionTestCase.testCrossContextSessionIncludeAccessTimeInvocation[proxy][http2]: java.lang.AssertionError
	at io.undertow.servlet.test.session.CrossContextServletSessionTestCase.testCrossContextSessionIncludeAccessTimeInvocation(CrossContextServletSessionTestCase.java:368)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:524)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:297)
------- Stderr: -------
14:10:29,741 DEBUG (XNIO-1 task-18) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /1/accesstimeservlet, authentication required: false
14:10:29,741 DEBUG (XNIO-1 task-18) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@7230281 for /1/accesstimeservlet
14:10:29,741 DEBUG (XNIO-1 task-18) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /1/accesstimeservlet
14:10:29,749 DEBUG (XNIO-1 task-9) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /1/include, authentication required: false
14:10:29,749 DEBUG (XNIO-1 task-9) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@7230281 for /1/include
14:10:29,749 DEBUG (XNIO-1 task-9) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /1/include
14:10:29,811 DEBUG (XNIO-1 task-1) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /1/include, authentication required: false
14:10:29,812 DEBUG (XNIO-1 task-1) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@7230281 for /1/include
14:10:29,812 DEBUG (XNIO-1 task-1) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /1/include
14:10:29,867 DEBUG (XNIO-1 task-5) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /1/include, authentication required: false
14:10:29,868 DEBUG (XNIO-1 task-5) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@7230281 for /1/include
14:10:29,868 DEBUG (XNIO-1 task-5) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /1/include
14:10:29,891 DEBUG (XNIO-1 task-22) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /1/accesstimeservlet, authentication required: false
14:10:29,891 DEBUG (XNIO-1 task-22) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@7230281 for /1/accesstimeservlet
14:10:29,892 DEBUG (XNIO-1 task-22) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /1/accesstimeservlet


io.undertow.websockets.jsr.test.annotated.AnnotatedEndpointTest.testEncodingAndDecodingText: java.lang.IllegalStateException: ChannelPipeline does not contain a HttpRequestEncoder or HttpClientCodec
	at io.netty.handler.codec.http.websocketx.WebSocketClientHandshaker$1.operationComplete(WebSocketClientHandshaker.java:182)
	at io.netty.handler.codec.http.websocketx.WebSocketClientHandshaker$1.operationComplete(WebSocketClientHandshaker.java:172)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:507)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:481)
	at io.netty.util.concurrent.DefaultPromise.access$000(DefaultPromise.java:34)
	at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:431)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
------- Stderr: -------
14:21:55,710 DEBUG (XNIO-1 task-15) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /ws/encoding/Stuart, authentication required: false
14:21:55,710 DEBUG (XNIO-1 task-15) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@204e50b1 for /ws/encoding/Stuart
14:21:55,711 DEBUG (XNIO-1 task-15) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /ws/encoding/Stuart


```

**Pull request merged**